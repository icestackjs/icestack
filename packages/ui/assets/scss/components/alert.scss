.alert {
  @apply rounded-box text-base-content border-base-200 border p-4;

  &-info {
    @apply text-info-content border-info/20 bg-info;
  }
  &-success {
    @apply text-success-content border-success/20 bg-success;
  }
  &-warning {
    @apply text-warning-content border-warning/20 bg-warning;
  }
  &-error {
    @apply text-error-content border-error/20 bg-error;
  }
}
