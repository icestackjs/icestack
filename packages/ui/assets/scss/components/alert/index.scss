@use "../../utils/index.scss" as *;
@use "sass:map";
$TypeMap: inject("alert.colors");
$DefaultMap: inject("alert.defaults.styled");

.alert {
  @include resolve(map.get($DefaultMap, "default"));

  @each $typeName, $map in $TypeMap {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
