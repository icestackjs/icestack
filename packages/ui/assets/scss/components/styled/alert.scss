@use "../utils/index.scss" as *;
@use "sass:map";
$TypeMap: injectAlertColors();
$DefaultMap: injectAlertDefaults();
.alert {
  @apply rounded-box border p-4;
  $default: map.get($DefaultMap, "default");
  @apply #{$default};
  @each $typeName, $map in $TypeMap {
    $d: map.get($map, "default");
    &#{getSelector($typeName)} {
      @apply #{$d};
    }
  }
}
