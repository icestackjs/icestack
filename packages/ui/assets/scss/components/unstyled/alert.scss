@use "../../utils/index.scss" as *;
@use "sass:map";
$DefaultMap: inject("alert.defaults.unstyled");

.alert {
  @include resolve(map.get($DefaultMap, "default"));
}
