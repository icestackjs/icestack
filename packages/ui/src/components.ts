export { componentNames as names, removeDefaultComponents, schemaMap } from '@icestack/preset-default/components'
