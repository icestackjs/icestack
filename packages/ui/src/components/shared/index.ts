export type * from '@icestack/shared'
export * from './utils'
