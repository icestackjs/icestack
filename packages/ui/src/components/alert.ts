import { Types, expandColorsMap, IDefaults } from './shared'

const colorsMap = expandColorsMap(Types, (typeName) => {
  return {
    default: `text-${typeName}-content border-${typeName}/20 bg-${typeName}`
  }
})

const defaults: IDefaults = {
  styled: {
    default: 'rounded-box border p-4 text-base-content border-base-200'
  },
  unstyled: {
    default:
      'grid w-full grid-flow-row content-start items-center justify-items-center gap-4 text-center sm:grid-flow-col sm:grid-cols-[auto_minmax(auto,1fr)] sm:justify-items-start sm:text-left'
  }
}
// const injectName = createInjectName('alert')
// const sassColors = transformJsToSass(colorsMap)
// const sassDefaults = transformJsToSass(defaults)
// export const inject = {
//   [injectName.colors]: () => {
//     return sassColors
//   },
//   [injectName.defaults]: () => {
//     return sassDefaults
//   }
// }

export const options = {
  colors: colorsMap,
  defaults
}
