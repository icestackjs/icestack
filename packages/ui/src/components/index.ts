import { GetSchemaFn, preprocessCssInJs } from './shared'
import collapse from './collapse'
import join from './join'
import indicator from './indicator'
import divider from './divider'
import stack from './stack'
import tab from './tab'
import avatar from '@/components/avatar'
import button from '@/components/button'
import alert from '@/components/alert'
import badge from '@/components/badge'
import chat from '@/components/chat'
import checkbox from '@/components/checkbox'
import input from '@/components/input'
import link from '@/components/link'
import progress from '@/components/progress'
import textarea from '@/components/textarea'
import toggle from '@/components/toggle'
import select from '@/components/select'
import radio from '@/components/radio'
import range from '@/components/range'
import loading from '@/components/loading'
import mask from '@/components/mask'
import form from '@/components/form'
import table from '@/components/table'
import skeleton from '@/components/skeleton'
import radialProgress from '@/components/radialProgress'
import countdown from '@/components/countdown'
import diff from '@/components/diff'
import kbd from '@/components/kbd'
import tooltip from '@/components/tooltip'
import toast from '@/components/toast'
import steps from '@/components/steps'
import dropdown from '@/components/dropdown'
import swap from '@/components/swap'
import card from '@/components/card'
import carousel from '@/components/carousel'
import stat from '@/components/stat'
import timeline from '@/components/timeline'
import breadcrumbs from '@/components/breadcrumbs'
import menu from '@/components/menu'
import bottomNavigation from '@/components/bottom-navigation'
import navbar from '@/components/navbar'
import fileInput from '@/components/file-input'
const schemaMap = {
  alert,
  avatar,
  button,
  badge,
  chat,
  checkbox,
  input,
  link,
  progress,
  textarea,
  toggle,
  select,
  radio,
  range,
  loading,
  mask,
  form,
  table,
  // tabs,
  skeleton,
  'radial-progress': radialProgress,
  countdown,
  diff,
  kbd,
  tooltip,
  toast,
  steps,
  collapse,
  join,
  indicator,
  divider,
  stack,
  tab,
  dropdown,
  swap,
  card,
  carousel,
  stat,
  timeline,
  breadcrumbs,
  menu,
  'bottom-navigation': bottomNavigation,
  navbar,
  'file-input': fileInput
} as const

const componentNames = Object.keys(schemaMap) as (keyof typeof schemaMap)[] // as unknown as keyof typeof _schemaMap

const resolvedSchemaMap = {} as Record<string, { schema: GetSchemaFn }>
for (const componentName of componentNames) {
  const o = schemaMap[componentName as keyof typeof schemaMap].schema
  resolvedSchemaMap[componentName] = {
    schema: (...args) => {
      const { defaults, selector } = o(...args)
      return {
        selector,
        defaults: preprocessCssInJs(defaults)
      }
    }
  }
}

// const names = Object.keys(resolvedSchemaMap) as (keyof typeof schemaMap)[]

const removeDefaultComponents = componentNames.reduce<Record<string, false>>((acc, cur) => {
  acc[cur] = false
  return acc
}, {})
export { componentNames as names, schemaMap, resolvedSchemaMap, removeDefaultComponents }

export { expandTypes, getSelector, compressCssSelector, preprocessCssInJs, transformCss2Js } from './shared'

export type { GetSchemaFn } from './shared'
