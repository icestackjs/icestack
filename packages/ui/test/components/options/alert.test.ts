import { handleOptions } from '@/components/shared'
import { options } from '@/components/alert'
import { getCodegenOptions } from '@/options'
describe.skip('alert', () => {
  it('merge options case 0', () => {
    const opts = handleOptions(
      options({
        options: getCodegenOptions(undefined, true),
        types: []
      }),
      {
        extend: undefined,
        override: undefined
      }
    )
    expect(opts).toMatchSnapshot()
  })

  it('merge options case 1', () => {
    const opts = handleOptions(
      options({
        options: getCodegenOptions(undefined, true),
        types: []
      }),
      {
        extend: undefined,
        override: {
          defaults: {
            styled: {
              default: {
                apply: []
              }
            },
            base: {
              default: {
                apply: ['bg-base-200'],
                css: {
                  color: 'red'
                }
              }
            }
          }
        }
      }
    )
    expect(opts).toMatchSnapshot()
  })

  it('merge options case 2', () => {
    const opts = handleOptions(
      options({
        options: getCodegenOptions(undefined, true),
        types: []
      }),
      {
        extend: {
          defaults: {
            styled: {
              default: {
                apply: []
              }
            },
            base: {
              default: {
                apply: ['bg-base-200'],
                css: {
                  color: 'red'
                }
              }
            }
          }
        },
        override: {
          defaults: {
            styled: {
              default: {
                apply: []
              }
            },
            base: {
              default: {
                apply: ['bg-base-200'],
                css: {
                  color: 'red'
                }
              }
            }
          }
        }
      }
    )
    expect(opts).toMatchSnapshot()
  })

  it('merge options case 3', () => {
    const opts = handleOptions(
      options({
        options: getCodegenOptions(undefined, true),
        types: []
      }),
      {
        extend: {
          defaults: {
            styled: {
              default: {
                apply: []
              }
            },
            base: {
              default: {
                apply: ['bg-base-200'],
                css: {
                  color: 'red'
                }
              }
            }
          }
        }
      }
    )
    expect(opts).toMatchSnapshot()
  })
})
