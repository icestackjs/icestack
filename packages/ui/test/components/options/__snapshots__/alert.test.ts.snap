// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`alert > merge options case 0 1`] = `
{
  "colors": {},
  "defaults": {
    "base": {
      "apply": [
        "grid",
        "w-full",
        "grid-flow-row",
        "content-start",
        "items-center",
        "justify-items-center",
        "gap-4",
        "text-center",
        "sm:grid-flow-col",
        "sm:grid-cols-[auto_minmax(auto,1fr)]",
        "sm:justify-items-start",
        "sm:text-left",
      ],
    },
    "styled": {
      "apply": [
        "rounded-box",
        "border",
        "p-4",
        "text-base-content",
        "border-base-400",
      ],
    },
  },
  "selector": ".alert",
}
`;

exports[`alert > merge options case 1 1`] = `
{
  "colors": {},
  "defaults": {
    "base": {
      "apply": [
        "grid",
        "w-full",
        "grid-flow-row",
        "content-start",
        "items-center",
        "justify-items-center",
        "gap-4",
        "text-center",
        "sm:grid-flow-col",
        "sm:grid-cols-[auto_minmax(auto,1fr)]",
        "sm:justify-items-start",
        "sm:text-left",
      ],
      "default": {
        "apply": [
          "bg-base-200",
        ],
        "css": {
          "color": "red",
        },
      },
    },
    "styled": {
      "apply": [
        "rounded-box",
        "border",
        "p-4",
        "text-base-content",
        "border-base-400",
      ],
      "default": {
        "apply": [],
      },
    },
  },
  "selector": ".alert",
}
`;

exports[`alert > merge options case 2 1`] = `
{
  "colors": {},
  "defaults": {
    "base": {
      "apply": [
        "grid",
        "w-full",
        "grid-flow-row",
        "content-start",
        "items-center",
        "justify-items-center",
        "gap-4",
        "text-center",
        "sm:grid-flow-col",
        "sm:grid-cols-[auto_minmax(auto,1fr)]",
        "sm:justify-items-start",
        "sm:text-left",
      ],
      "default": {
        "apply": [
          "bg-base-200",
          "bg-base-200",
        ],
        "css": {
          "color": "red",
        },
      },
    },
    "styled": {
      "apply": [
        "rounded-box",
        "border",
        "p-4",
        "text-base-content",
        "border-base-400",
      ],
      "default": {
        "apply": [],
      },
    },
  },
  "selector": ".alert",
}
`;

exports[`alert > merge options case 3 1`] = `
{
  "colors": {},
  "defaults": {
    "base": {
      "apply": [
        "grid",
        "w-full",
        "grid-flow-row",
        "content-start",
        "items-center",
        "justify-items-center",
        "gap-4",
        "text-center",
        "sm:grid-flow-col",
        "sm:grid-cols-[auto_minmax(auto,1fr)]",
        "sm:justify-items-start",
        "sm:text-left",
      ],
      "default": {
        "apply": [
          "bg-base-200",
        ],
        "css": {
          "color": "red",
        },
      },
    },
    "styled": {
      "apply": [
        "rounded-box",
        "border",
        "p-4",
        "text-base-content",
        "border-base-400",
      ],
      "default": {
        "apply": [],
      },
    },
  },
  "selector": ".alert",
}
`;
